Pk/feat mail service

pkomesu opened the pull request:
Este PR contiene: 
- Envío de e-mail de reserva provisoria con pago en cash.
- Envío de e-mail de reserva provisoria con pago por MP.
- Envío de e-mail de confirmación de pago en cash.
- Envío de e-mail de confirmación de pago por MP.

Commit:
feat: EmailDto

Se crea la clase EmailDto para enviar e-mails.

Commit:
feat: interfaz IMailingService

Se crea interfaz IMailingService para envío de e-mails.

Commit:
feat: campos en application*.yaml para mailing

Se agrega a application.yaml y application-prod.yaml campos para envío de e-mails.

Commit:
feat: dependencia para mailing

Se agrega a archivo pom.xml dependencia para envío de e-mails.

Commit:
feat: TransportMailSenderImpl

Se implementa TransportMailSenderImpl para envío de e-mails utilizando dependencias del paquete jakarta.mail.

Commit:
feat: endpoint para envío de e-mails

Se agrega endpoint para envío de e-mails.

Commit:
docs: actualizo changelog

Commit:
feat: JavaMailSenderImpl

Se implementa clase JavaMailSenderImpl para envío de e-mails simples con solo texto.

Commit:
docs: actualizo changelog

Commit:
fix: se utiliza JavaMailSenderImpl para endpoint de test

fix: Se utiliza JavaMailSenderImpl para endpoint de test de envío de e-mails.

Commit:
docs: actualizo changelog

Commit:
fix: se reemplaza IMailingService por AbstractMailingService

Se elimina IMailingService y se reemplaza por AbstractMailingService.

Commit:
fix: mejoras en notificación por e-mail de reserva provisoaria

Se realizan mejoras en la notificación por e-mail de reservas provisorias.

Commit:
feat: se agrega env PAYMENT_MP_ACCOUNT para informar cuenta de Mercado Pago

Se agrega a deployment-prod.yaml env PAYMENT_MP_ACCOUNT para informar por e-mail la cuenta de Mercado Pago.

Commit:
fix: se agrega envío de e-mails a demanda para testing

Se agrega a ContactController el envío a demanda de e-mails para testing.

Commit:
docs: actualizo changelog

Commit:
feat: templates para envío de emails en formato texto

Se agregan templates para envío de emails en formato texto.

Commit:
fix: se agregan envs con datos del evento

Se agregan envs EVENT_ADDRESS, EVENT_DATE_TIME, EVENT_NAME y EVENT_PLACE para cargar los datos del evento.

Commit:
fix: se agrega uso de envs en AbstractMailingService

Se agrega uso de envs con datos del evento en AbstractMailingService.

Commit:
docs: modifico descripción de atributos en JavaMailSenderImpl

docs: Se modifica la descripción de atributos en la clase JavaMailSenderImpl.

Commit:
fix: se mejora el cuerpo html del email de reserva provisoria

fix: se mejora el cuerpo html del email de reserva provisoria.

Commit:
fix: se agrega test para obtención de templates html

Se agregan test unitarios para obtención de templates html para emails de reserva provisoria.

Commit:
feat: se agregan templates html para emails de reserva provisoria

Se agregan templaes html para emails de reserva provisoria.

Commit:
fix: se modifica ContactController para test de emails

fix: Se modificaa clase ContactController para el test de envío a demanda de e-mails con formato html.

Commit:
docs: actualizo changelog

Commit:
fix: se agregan envs MAIL_USERNAME y MAIL_PASSWORD

Se agrega uso de envs MAIL_USERNAME y MAIL_PASSWORD.

Commit:
docs: actualizo changelog

Commit:
fix: se separa eventDateTime en dos envs

Se separa env eventDateTime en dos envs eventDate y eventTime.

Commit:
fix: se agregan envs EVENT_DATE y EVENT_TIME

Se agrega a application*.yaml envs EVENT_DATE y EVENT_TIME.

Commit:
feat: Se agrega mapa para carga de imagenes al template html

Se agrega a EmailDto el uso de mapa para carga de imagenes al template html.

Commit:
feat: se implementa carga de imagenes al template html

feat: Se implementa carga de imagenes al template html.

Commit:
feat: se implementa metodo abstracto notifyConfirmation

feat: Se implementa en TransportMailSenderImpl método abstracto notifyConfirmation para envío de e-mails de confirmación de pago.

Commit:
fix: se formatean templates html para reserva provisoria

fix: Se formatean templates html para reserva provisoria.

Commit:
feat: se agregan templates html para confirmación de pago

Se agregan templates html para confirmación de pago.

Commit:
fix: se agrega imagen .png con el header del evento

fix: Se agrega imagen .png para ser enviado como header en los e-mails de reserva y confirmación de pago.

Commit:
fix: se agrega endpoint para testing de e-mails de confirmacion de pago

Se agrega endpoint para testing de e-mails de confirmación de pago.

Commit:
docs: actualizo changelog

Commit:
fix: merge conflicts

keynaka commented on src/main/resources/mailing/templates/text/reservation-cash-template.txt:
```diff
@@ -0,0 +1,14 @@
+Hola ${NAME} ${LASTNAME}!
```
Estos archivos .txt quedaron de mas?

pkomesu replied:
Estos habían quedado porque la primera implementación de envío de e-mails fue con solo texto. Para esta implementación había usado la dependencia JavaMail.
Después cuándo avanzamos a envío de e-mails con html, JavaMail no lo permitía. 
Ya los removí.

keynaka commented on src/main/java/com/rkmd/toki_no_nagare/controller/ContactController.java:
```diff
@@ -39,4 +56,94 @@ public ResponseEntity<Contact> createContact(@RequestBody @Valid Map<String, Obj
+    /** Uso provisorio. Eliminar este endpoint al finalizar los tests de envío de e-mails. */
+    @Deprecated
+    @GetMapping("/test/notifications/reservations/{email}/{paymentMethod}/{name}/{lastname}")
+    public ResponseEntity<String> testSendReservationMail(@PathVariable("email") String email,
```
ya estos metodos no son tan tests que digamos no?

pkomesu replied:
Los endpoints no se utilizarían en prod. 
Los dejo por ahora por alguien quiere hacer pruebas de modificar los e-mails.

keynaka commented on src/main/java/com/rkmd/toki_no_nagare/controller/ContactController.java:
```diff
@@ -39,4 +56,94 @@ public ResponseEntity<Contact> createContact(@RequestBody @Valid Map<String, Obj
+    /** Uso provisorio. Eliminar este endpoint al finalizar los tests de envío de e-mails. */
+    @Deprecated
+    @GetMapping("/test/notifications/confirmations/{email}/{paymentMethod}/{name}/{lastname}")
+    public ResponseEntity<String> testSendConfirmationMail(@PathVariable("email") String email,
```
idem

pkomesu replied:
Los endpoints no se utilizarían en prod. 
Los dejo por ahora por alguien quiere hacer pruebas de modificar los e-mails.

Commit:
feat: se agrega template html para e-mail de expiracion

Se agrega template html para envío de e-mail de expiración de reserva.

Commit:
feat: se agrega metodo para notificar expiracion de reserva

Se agrega método para notificar expiración de reserva por e-mail.

Commit:
fix: se agrega endpoint para test de notificacion de expiracion

fix: Se agrega endpoint para testing de e-mails de expiración de reserva.

Commit:
docs: actualizo changelog

Commit:
fix: se remueven templates de e-mail en formato .txt

Se remueven templates de e-mails en formato .txt.

Commit:
test: se agregan test de carga de templates html para confirmacion y expiracion

test: Se agregan test de carga de templates html para confirmacion y expiracion.

Commit:
feat: se agrega metodo abstracto sendSimpleMail

feat: Se agrega a AbstractMailingService método abstracto para enviar e-mail en formato texto.

Commit:
docs: actualizo changelog

Commit:
fix: se elimina implementacion JavaMailSenderImpl

fix: Se remueve implementacion JavaMailSenderImpl.

Commit:
fix: se remueve atributo sin uso en ContactController

Commit:
test: se corrigen test en TransportMailSenderImplTest

test: se corrigen test en TransportMailSenderImplTest.

Commit:
docs: actualizo changelog

keynaka commented:
Listo, quedan los conflictos de merge y dale mecha nomas!

Commit:
fix: conflictos de merge
